utilities/set-up-error-reporting-for-scripts.sh

```diff
@@ -1,12 +1,12 @@
 #!/usr/bin/env bash
-set -euo pipefail
+set -eEuo pipefail
```

Comment:
Is there a reason why we're not including the `-E` flag generally? If an error is thrown in a function, our cleanup function actually doesn't catch it properly and treats it as the `EXIT` trap code instead of `ERR`. Found that out in this PR

https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html

Reply:
> Is there a reason why we're not including the `-E` flag generally?

No good one that I know of. :)